cmd/wfx/cmd/root/plugins.go

```diff
@@ -0,0 +1,68 @@
+package root
+
+/*
+ * SPDX-FileCopyrightText: 2023 Siemens AG
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Author: Michael Adler <michael.adler@siemens.com>
+ */
+
+import (
+	"os"
+	"path"
+	"path/filepath"
+	"sort"
+
+	"github.com/Southclaws/fault"
+	"github.com/rs/zerolog/log"
+	"github.com/siemens/wfx/middleware"
+	"github.com/siemens/wfx/middleware/plugin"
+)
+
+func createPluginMiddlewares(pluginsDir string, quitChan chan error) ([]middleware.IntermediateMW, error) {
+	pluginMws, err := loadPlugins(pluginsDir)
+	if err != nil {
+		return nil, fault.Wrap(err)
+	}
+	result := make([]middleware.IntermediateMW, 0, len(pluginMws))
+	for _, p := range pluginMws {
+		mw, err := plugin.NewMiddleware(p, quitChan)
+		if err != nil {
+			return nil, fault.Wrap(err)
+		}
+		result = append(result, mw)
+	}
+	return result, nil
+}
+
+func loadPlugins(dir string) ([]plugin.Plugin, error) {
+	log.Debug().Msg("Loading plugins")
+	entries, err := os.ReadDir(dir)
+	if err != nil {
+		return nil, fault.Wrap(err)
+	}
+
+	result := make([]plugin.Plugin, 0, len(entries))
+	for _, entry := range entries {
+		if !entry.IsDir() {
+			dest, err := filepath.EvalSymlinks(path.Join(dir, entry.Name()))
+			if err != nil {
+				return nil, fault.Wrap(err)
+			}
+			info, err := os.Stat(dest)
+			if err != nil {
+				return nil, fault.Wrap(err)
+			}
+			// check if file is executable
+			if (info.Mode() & 0o111) != 0 {
+				result = append(result, plugin.NewFBPlugin(dest))
+			} else {
+				log.Warn().Str("dest", dest).Msg("Ignoring non-executable file")
```

Comment:
Hm, if there's a non-executable file found, this may hint to a misconfiguration, hence an error and fail-exit should follow. Or it is a feature to selectively enable/disable plugins.... What do we opt for (and document)?

Reply:
It could also be a configuration file or README for the plugin (in the same directory, to keep things together). I have added a note in the documentation.